languages/ruby/lib/oso/polar/ffi/polar.rb

```diff
@@ -51,6 +57,15 @@ def validate_roles_config(config)
           handle_error if result.zero?
         end
 
+        def build_filter_plan(types, partials, variable, class_tag)
+          types = JSON.dump(types)
+          partials = JSON.dump(partials)
+          plan = Rust.build_filter_plan(self, types, partials, variable, class_tag)
```

Comment:
no reason except that's not how we do it in python. maybe something to do next cycle

Reply:
Ok having thought about it a little, I think an FFI class would be overkill since the filter plan is just data with no Rust code it needs to call. But I did add some classes to separate the json wrangling from the actual filter logic, lmk if it's better